src/main/java/sirius/kernel/nls/NLS.java

```diff
@@ -1169,6 +1190,38 @@ private static <V> V parseBasicTypesFromUserString(Class<V> clazz, String value,
         return parseDatesFromUserString(clazz, value, language);
     }
 
+    private static Integer parseIntegerFromUser(String value, String language) {
+        try {
+            Double valueAsDouble = parseDecimalNumberFromUser(value, language);
```

Comment:
Probably I miss something, but isn't this exactly the code above, just nested in another catch clause? Why not invoke this?

And if it is better to copy the code, why not fire the desired exception class directly?

Reply:
There are basically 4 variants, slightly close but not the same.
- Integer x Long - different internal types
- FromUserString x FromMachineString - for the first we reuse the parseDoubleFromUserString, which throws a different exception as the Java's valueOf used in FromMachineString